### PR TITLE
refactor: eliminate duplicate code in projects

### PR DESCRIPTION
**Pull request by geigerzaehler**
Before, the functions `refresh` and `fetchPeers` used exactly the same code.

### CODE REVIEW

File: ui/Screen/Project.svelte [hunk 1]
```diff
@@ -46,13 +51,13 @@
     selectPeer(peer);
   };
 
-  onMount(() => {
+  onDestroy(
```
Comment (rudolfs) [inline ui/Screen/Project.svelte:54]: Why did you change this?

Comment (geigerzaehler) [reply ui/Screen/Project.svelte:54]: Because we did not return anything from the callback to `onMount` we never stopped listening to the events. I could have added a return but felt that `onDestroy` is more straight-forward.

Comment (rudolfs) [reply ui/Screen/Project.svelte:54]: Can we make this one more explicit as well?

Comment (geigerzaehler) [reply ui/Screen/Project.svelte]: https://github.com/radicle-dev/radicle-upstream/pull/1911/commits/6e2f1b544e8b6c807605520e572978995bd131fa

**Review by rudolfs — APPROVED** [review]
utACK

**Review by rudolfs — DISMISSED** [review]
